# Naive Bayes

### Simple but powerful algo with a lot of real world applications


### Prelim Knowledge


* **Conditional Probability** - Probability of an event A given the other event has occured. e.g. what is the probability that it will rain given that it is cloudy?


* **Joint Probability** - a measure that calculates likelihood of two or more events at the same time.


* **Proportionality** - relationship b/w 2 quantities that are related by multiplication to a constant.


* **Bayes Theorem** - In probability theory and statistics, Bayes' theorem (alternatively Bayes's theorem, Bayes's law or Bayes's rule) describes the probability of an event, based on prior knowledge of conditions that might be related to the event.


### Baye's Theorem
* **P (A | B) = P (B | A) * P(A) / P(B)**

### If we had Target (y) and Feature (X) variables... 


* **P (y | X) = P (X | y) * P(y) / P(X)**


### If I have more than one X, (X1, X2, X3)


* **P (X | y) = P(X1 | y) * P(X2 | y) * P(X3 | y)**
    - Baye's Theorem
        - **P (y | X) = P(X1 | y) * P(X2 | y) * P(X3 | y) * P(y) / P(X1) * P(X2) * P(X3)**

# Dataset : Depending on weather condition whether to play golf or not

<img src="img/naivebayes.png">

### Step 1 # Frequency Table

<img src='img/freq-table.png'>

### Step 2 # Ratio Table

<img src='img/ratio-table.png'>

### X  = { outlook : sunny,
###           temperature: mild,
###           humidity: normal,
###           windy: false
###         }

### P (yes | X)

#### P (yes) = 9/14

#### P (outlook = sunny | yes) = 2/9

#### P (temperature = mild | yes) = 4/9


#### P (humidity = normal | yes) = 6/9


#### P(windy = false | yes) = 6/9

In [1]:
# P (yes | X)

(9/14) * (2/9) * (4/9) * (6/9) * (6/9)

0.02821869488536155

#### P (yes | X) = 0.02821869488536155

#### P (no | X) = 0.0069

##### Since P (yes | X) > P(no | X), so we predict that person would play golf.

## Types of Naive Bayes

* Multinomial - Assumption is P(X | y) follows a multinomial distribution. Used frequently in Text Classification kind of 


* Gaussian 


* Bernoulli